test/unit/tools/test_wrappers.py

```diff
@@ -83,17 +85,40 @@ def test_raw_object_wrapper():
     assert not false_wrapper
 
 
+def valuewrapper(tool, value, paramtype):
+    if paramtype == "integer":
+        parameter = IntegerToolParameter(tool, XML('<param name="blah" type="integer" value="10" min="0" />'))
+    elif paramtype == "text":
+        parameter = TextToolParameter(tool, XML('<param name="blah" type="text" value="10"/>'))
+    elif paramtype == "float":
+        parameter = FloatToolParameter(tool, XML('<param name="bla" type="float" value="10"/>'))
+    return InputValueWrapper(parameter, str(value))
+
+
 @with_mock_tool
-def test_input_value_wrapper(tool):
-    parameter = IntegerToolParameter(tool, XML('<param name="blah" type="integer" value="10" min="0" />'))
-    wrapper = InputValueWrapper(parameter, "5")
+def test_input_value_wrapper_comparison(tool):
+    wrapper = valuewrapper(tool, 5, "integer")
     assert str(wrapper) == "5"
     assert int(wrapper) == 5
     assert wrapper == "5"
     assert wrapper == 5
     assert wrapper == 5.0
     assert wrapper > 2
     assert wrapper < 10
+    assert wrapper <= 5.1
+
+
+@with_mock_tool
+def test_input_value_wrapper_input_value_wrapper_comparison(tool):
+    wrapper = valuewrapper(tool, 5, "integer")
+    assert str(wrapper) == valuewrapper(tool, "5", "text")
+    assert int(wrapper) == valuewrapper(tool, "5", "integer")
+    assert wrapper == valuewrapper(tool, "5", "text")
```

Comment:
I think this should not be allowed, too much magic, even in normal Python `"5" == 5` is False.

Reply:
But that has previously worked, see the test above that I didn't modify.

Reply:
that said this naturally falls out of casting InputValueWrapper to InputValueWrapper.input.type (https://github.com/galaxyproject/galaxy/pull/9047/commits/155660083d66d00423c2ed5ef6ee9028badd4883), which maybe means we should do this ?